src/main/java/org/jenkinsci/plugins/docker/commons/DockerRegistryEndpoint.java

```diff
@@ -54,6 +71,36 @@ public DockerRegistryEndpoint(String url, String credentialsId) {
     }
 
     /**
+     * Parse the registry endpoint out of a registry:port/namespace/name:tag string. Credentials are set to the id
+     * passed. The url is built from registry:port into https://registry:port, the same way docker push does.
+     * 
+     * @param s
+     * @param credentialsId
+     *            can be null
+     * @throws IllegalArgumentException
+     *             if string can't be parsed
+     * @return The DockerRegistryEndpoint corresponding to the registry:port part of the string
+     */
+    public static DockerRegistryEndpoint parse(String s, @CheckForNull String credentialsId) {
```

Comment:
`parse` is a pretty generic name, and the input string here is not really a representation of the endpoint. It is an image name, as constructed by the `imageName` method. So I would suggest calling this `fromImageName`, and adding bidirectional Javadoc links with `imageName` to make it clear that these are related operations.